tools/bin/ci_integration_workflow_launcher.sh

```diff
@@ -1,42 +1,102 @@
 
-if [ "$MATCHING_WORKFLOW_IDS" -ne "1" ]; then
-  echo "More than one workflow exists with the path $WORKFLOW_PATH"
+workflow_id=$(echo $workflow_ids_curl_response | \
+    jq '.workflows[] | select(.path==".github/workflows/test-command.yml") | .id')
+
+# We expect a unique response, 2 responses is too much
+workflows_with_matching_paths=$(echo $workflow_id | wc -l)
+echo -e "$blue_text""jq returned: $workflow_id""$default_text"
+
+if test "$workflows_with_matching_paths" -ne "1" ; then
+  echo -e "\n\n$red_text""Unexpected number of workflows found != 1 for .github/workflows/test-command.yml""$default_text"
+  echo -e "\n\n$red_text""\$workflows_with_matching_paths = $workflows_with_matching_paths""$default_text"
   exit 1
 fi
 
-MAX_RUNNING_MASTER_WORKFLOWS=5
-RUNNING_MASTER_WORKFLOWS=$(curl "$REPO_API/actions/workflows/$WORKFLOW_ID/runs?branch=master&status=in_progress" --header "Authorization: Bearer $GITHUB_TOKEN" | jq -r ".total_count")
-if [ "$RUNNING_MASTER_WORKFLOWS" -gt "$MAX_RUNNING_MASTER_WORKFLOWS" ]; then
-  echo "More than $MAX_RUNNING_MASTER_WORKFLOWS integration tests workflows running on master."
-  echo "Skipping launching workflows."
+# --------- Ensure no more than 5 concurrent tests happen
+
+max_running_master_workflows=5
+
+running_master_workflows_curl_response=$(
+  curl \
+    --silent \
+    --show-error \
+    --header "Authorization: Bearer $GITHUB_TOKEN" \
+    --request GET "$repo_api/actions/workflows/$workflow_id/runs?branch=master&status=in_progress")
+
+echo -e "$blue_text""\$running_master_workflows_curl_response is \n\n$running_master_workflows_curl_response\n\n""$default_text"
+echo -e "$blue_text""Running jq on response jq .total_count""$default_text"
+
+running_master_workflows=$(echo "$running_master_workflows_curl_response" | jq .total_count)
+echo -e "$blue_text""JQ Found \"$running_master_workflows\" running""$default_text"
+
+
+if test "$running_master_workflows" -gt "$max_running_master_workflows"; then
+  echo -e "$red_text""More than $max_running_master_workflows integration tests workflows running on master.""$default_text"
+  echo -e "$red_text""Skipping launching workflows.  If you want this test run use manual steps!""$default_text"
   exit 0
+else
+  echo -e "$blue_text""Running ./gradlew integrationTest --dry-run""$default_text"
 fi
 
-CONNECTORS=$(./gradlew integrationTest --dry-run | grep 'integrationTest SKIPPED' | cut -d: -f 4 | sort | uniq)
-echo "$CONNECTORS" | while read -r connector; do
-  if [ ! -z "$connector" ]; then
-    echo "Issuing request for connector $connector..."
-    curl \
-      -i \
-      -X POST \
-      -H "Accept: application/vnd.github.v3+json" \
-      -H "Authorization: Bearer $GITHUB_TOKEN" \
-      "$REPO_API/actions/workflows/$WORKFLOW_ID/dispatches" \
-      -d "{\"ref\":\"master\", \"inputs\": { \"connector\": \"$connector\"} }"
-  fi
+# --------- Use gradle to find tests to fire
+
+./gradlew integrationTest --dry-run
```

Comment:
do we need to run gradle twice to figure out if it works?

Reply:
I tried capturing the output, but it did really weird stuff with newlines.  I honestly think it has something to do with being an interactive shell or not, but I got tired of messing with it and just ran the command twice so humans could see an error and I could still capture the output correctly for cut... which is more picky about proper newlines